src/PowerShellEditorServices/Services/PowerShellContext/Utilities/CommandHelpers.cs

```diff
@@ -16,20 +16,26 @@ namespace Microsoft.PowerShell.EditorServices.Services.PowerShellContext
     /// </summary>
     internal static class CommandHelpers
     {
-        private static readonly ConcurrentDictionary<string, bool> NounExclusionList =
+        private static readonly ConcurrentDictionary<string, bool> s_nounExclusionList =
             new ConcurrentDictionary<string, bool>();
 
+        private static readonly ConcurrentDictionary<string, CommandInfo> s_commandInfoCache =
+            new ConcurrentDictionary<string, CommandInfo>();
+
+        private static readonly ConcurrentDictionary<string, string> s_synopsisCache =
+            new ConcurrentDictionary<string, string>();
+
         static CommandHelpers()
         {
-            NounExclusionList.TryAdd("Module", true);
-            NounExclusionList.TryAdd("Script", true);
-            NounExclusionList.TryAdd("Package", true);
-            NounExclusionList.TryAdd("PackageProvider", true);
-            NounExclusionList.TryAdd("PackageSource", true);
-            NounExclusionList.TryAdd("InstalledModule", true);
-            NounExclusionList.TryAdd("InstalledScript", true);
-            NounExclusionList.TryAdd("ScriptFileInfo", true);
-            NounExclusionList.TryAdd("PSRepository", true);
+            s_nounExclusionList.TryAdd("Module", true);
+            s_nounExclusionList.TryAdd("Script", true);
+            s_nounExclusionList.TryAdd("Package", true);
+            s_nounExclusionList.TryAdd("PackageProvider", true);
+            s_nounExclusionList.TryAdd("PackageSource", true);
+            s_nounExclusionList.TryAdd("InstalledModule", true);
+            s_nounExclusionList.TryAdd("InstalledScript", true);
+            s_nounExclusionList.TryAdd("ScriptFileInfo", true);
+            s_nounExclusionList.TryAdd("PSRepository", true);
```

Comment:
According to the comment on line 62, shouldn't `PowerShellGet` be in this list?

Reply:
Ah nevermind, this is the noun list, duh.  Not the package name list.

Reply:
That said, I see more nouns from `PowerShellGet` 2.2.3 than appear in the list above.  `gcm -m PowerShellGet | % Noun | % Sort -uniq`  Is that a problem?

Reply:
Yeah I just renamed the variable. We should have a better check for verb-noun when PSGet uses the same noun as an inbox module `Module` as an example.

Reply:
I've added a s_cmdletExclusionList now which will handle that ^